Handle writing automatically

forslund opened the pull request:
Using decorator as suggested by @MatthewScholefield. The decorator sets
an internal attirbute so following calls to a decorated function will
not try to write data.


Apparently I didn't push this commit last week. Lets add it as well to keep the python interface the same.

forslund committed:
Handle writing automatically

Using decorator as suggested by @MatthewScholefield. The decorator sets
an internal attirbute so following calls to a decorated function will
not try to write data.

MatthewScholefield commented on msm/mycroft_skills_manager.py:
```diff
@@ -207,12 +226,16 @@ def update(self, skill=None, author=None):
             try:
-                func(skill)
+                if 'save_skills_data' in dir(func):
+                    func(skill, __msm_write=False)
```
Does the `__msm_write=False` still refer to anything?

forslund replied:
No, you're right. That's a remnant from my first approach. Removing

forslund commented:
There, the code should be cleaner now.

MatthewScholefield reviewed (approved):
Looks good!

MatthewScholefield reviewed (changes requested):
Just thought of an edge case where this could mess up:
If the function throws an exception I don't think it saves skills (probably desired anyways) but also might not reset the `saving_handled` variable. For example:
```python
try:
    msm.install(skill_name, skill_author)
except SkillNotFound:
    msm.install(skill_name)  # self.saving_handled already True but won't save
```

forslund commented:
Ah good thinking. I'll think it through and do an update.

A try, finally might be enough to handle it.

forslund committed:
Make sure saving_handled is reset

Saving handled flag will always be reset even if an exception occurs. Write will still only occur if the msm operation was successful.
